go/src/koding/klient/client/client_test.go

```diff
@@ -19,12 +19,63 @@ func (f mockCaller) Call(v ...interface{}) error {
 	return f(v)
 }
 
+// handlerWrapper is a kite Handler middle-ware that allows to check when
+// handling function finished its execution.
+func handlerWrapper(h kite.HandlerFunc) (<-chan struct{}, kite.HandlerFunc) {
+	doneC := make(chan struct{}, 1)
+	return doneC, kite.HandlerFunc(func(req *kite.Request) (interface{}, error) {
+		result, err := h(req)
+		select {
+		case doneC <- struct{}{}:
+		case <-time.After(5 * time.Second):
+			panic("invalid number of handler callas")
+		}
+
+		return result, err
+	})
+}
+
+// wait is a helper function that waits for an event from done channel or
+// timeouts after specified timeout. This function must be called from main
+// go-routine only.
+func wait(t *testing.T, doneC <-chan struct{}, timeout time.Duration) {
```

Comment:
> this will add ~50-60 lines to this test

Don't bother, we have a lot of lines, can throw more if needed 😎 

![pic](https://m.popkey.co/30ad0d/1zrRX.gif)